pkg/storage/stores/shipper/bloomshipper/shipper.go

```diff
@@ -125,19 +141,40 @@ func (s *Shipper) findBlocks(
 	return blockRefs
 }
 
-func isOutsideRange(
-	b *BlockRef,
-	minFingerprint uint64,
-	maxFingerprint uint64,
-	startTimestamp int64,
-	endTimestamp int64,
-) bool {
-	return b.MaxFingerprint < minFingerprint || b.MinFingerprint > maxFingerprint ||
-		b.EndTimestamp < startTimestamp || b.StartTimestamp > endTimestamp
+// getPosition returns the index of element v in slice s when v >= elem
+// TODO(chaudum): Use binary search to find index instead of iteration.
+func getPosition[S ~[]E, E cmp.Ordered](s S, v E) int {
+	for i := range s {
+		if v > s[i] {
+			continue
+		}
+		return i
+	}
+	return len(s)
+}
+
+func isOutsideRange(b *BlockRef, startTimestamp, endTimestamp int64, fingerprints []uint64) bool {
+	// First, check time range
+	if b.EndTimestamp < startTimestamp || b.StartTimestamp > endTimestamp {
+		return true
+	}
+
+	// Then, check if outside of min/max of fingerprint slice
+	minFp, maxFp := getFromThrough(fingerprints)
+	if b.MaxFingerprint < minFp || b.MinFingerprint > maxFp {
+		return true
+	}
+
+	// Check if the block range is inside a "gap" in the fingerprint slice
+	// e.g.
+	// fingerprints = [1, 2,          6, 7, 8]
+	// block =              [3, 4, 5]
+	idx := getPosition[[]uint64](fingerprints, b.MinFingerprint)
+	return b.MaxFingerprint < fingerprints[idx]
```

Comment:
this check should give a `true` if no one item from `fingerprints []uint64` is greater or equal to `b.MinFingerprint` and less or equal to `b.MaxFingerprint`. 
however, this method returns `false` for the block `b.MinFingerprint: 3 , b.MaxFingerprint: 5` if fingerprints []uint64 is :
1.  `[2]` - should be true because the block does not cover this fingerprint
2. `[6]` - should be true because the block does not cover this fingerprint

could you please add more test cases to cover this method?

Reply:
These cases are actually covered in line 163-166